Src/StackifyLib/Internal/Auth/Claims/AppClaimsBuilderFullFramework.cs

```diff
@@ -187,12 +187,21 @@ private void SetWindowServiceAppName()
             catch (Exception ex)
             {
                 StackifyAPILogger.Log("Unable to get windows service name\r\n" + ex.ToString(), true);
-            }            
+            }
         }
 
         private async Task SetDeviceName()
         {
-            AppClaims.DeviceName = await GetEC2InstanceId() ?? Environment.MachineName;
+            var machineName = Environment.MachineName;
+
+            if (Config.IsEc2 == null || Config.IsEc2 == true || (machineName.StartsWith("EC2") && machineName.Contains("-")))
```

Comment:
For some reason I really want to keep this code isolated and not introduce it into more of the stack...it's worthy of further consideration though...